### ISSUE TITLE
Refactor requests to fetch dictionaries

### ISSUE THREAD

**Issue by kytta** [vavilon-js/vavilon.js#4]
**TL;DR**: I am not an expert in JS, so I kinda have no idea how to make files load in background to be used later

# How it is done now

Right now the script loading function is a `Promise`-ifyed XHR request. I decided to use it to decrease the bundle size, since XHR is in every browser.

Once the page is loaded to the point where one can read the dictionary files, the script starts to download them. At first I decided to store `Promise`s for each dictionary, but then I found out that the values will remain as Promises (duh). So I left it the not-so-practical way: every dictionary now is loaded with `await`, so the script won't execute until they're all done.

# How I want it to work

I want the function to work according to this workflow:

1. The script starts to be executed; the `pageLocale` and `userLocale` become determined.
2. The dictionary locales and urls are loaded. The script searches for the dictionary, whose locale (or the language half of it) match the `userLocale`
   - If such dictionary is found, it starts to load immediately. Script waits for it to load and saves its locale to `useDict`
   - If such dictionary is **not** found, the script continues to work. Every other dictionary start to load in the background
3. Script looks for vavilon-enabled elements
4. If the `useDict` is set, the script replaces the text inside vavilon-enabled elements

My problem is that right now I have no idea how to implement this. Might have to dig very deep for this. Any help with this will be highly appreciated

**Comment by kytta** [comment]
Okay, I might have figured it out after all 😄. I guess it can all be done with `XMLHttpRequest` and its `async` without any problems. Let's see...
